src/pageComponents/signPage/SignInForm/index.tsx

```diff
@@ -0,0 +1,106 @@
+import cx from 'classnames';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useForm } from 'react-hook-form';
+import { useState } from 'react';
+import { useRouter } from 'next/navigation';
+import nookies from 'nookies';
+
+import { schema } from './schema';
+import { useAuth } from '../../../context/AuthContext';
+import { authUser } from '../../../api/user';
+import { Loading } from '../../../components/ui/Loading';
+import { AriaLiveErrorMessage } from '../../../components/AriaLiveErrorMessage';
+import { Input } from '../../../components/Input';
+import { SECONDS_IN_WEEK, Routes } from '../../../utils/constants';
+import { FormData } from '../../../types';
+
+export const SignInForm = () => {
+  const {
+    register,
+    watch,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({
+    resolver: yupResolver(schema),
+    defaultValues: {
+      email: '',
+      password: '',
+    },
+  });
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isVisiblePassword, setVisiblePassword] = useState(false);
+
+  const router = useRouter();
+
+  const { login } = useAuth();
+
+  const onSubmit = async (formData: FormData) => {
+    setIsLoading(true);
+
+    try {
+      const response = (await authUser(formData, (error: any) =>
+        setErrorMessage(error.message),
+      )) as { token: string };
```

Comment:
The fact you need the casting here is fishy. Some of the types does probably not work

Reply:
😭 no idea where the fish is.